tests/Twig/Extension/SonataAdminExtensionTest.php

```diff
@@ -2116,6 +2051,131 @@ class="sonata-readmore"
         ];
     }
 
+    /**
+     * @group legacy
+     * @assertDeprecation Accessing a non existing value is deprecated since sonata-project/sonata-admin-bundle 3.x and will throw an exception in 4.0.
+     *
+     * @dataProvider getRenderViewElementWithNoValueTests
+     */
+    public function testRenderViewElementWithNoValue(string $expected, string $type, array $options): void
+    {
+        $this->admin
+            ->method('getTemplate')
+            ->willReturn('@SonataAdmin/CRUD/base_show_field.html.twig');
+
+        $this->fieldDescription
+            ->method('getValue')
+            ->willThrowException(new NoValueException());
+
+        $this->fieldDescription
+            ->method('getType')
+            ->willReturn($type);
+
+        $this->fieldDescription
+            ->method('getOptions')
+            ->willReturn($options);
+
+        $this->fieldDescription
+            ->method('getTemplate')
+            ->willReturnCallback(static function () use ($type) {
```

Comment:
Could we use `null` as default value instead?
```suggestion
            ->willReturnCallback(static function () use ($type): ?string {
```

Reply:
Sure, it's done.